doc/usecases/core/UC_keyset_insert.md

```diff
@@ -0,0 +1,29 @@
+# Use Case: Insert `Key` into `KeySet`
+
+## Summary
+
+- **Title:** Insert `Key` into `KeySet`
+- **Scope:** `libelektra-core`
+- **Level:** Developer Goal
+- **Actors:** Core, Caller
+- **Brief:** Caller inserts new `Key` into existing `KeySet`
+
+## Scenarios
+
+- **Precondition:**
+  - [`KeySet` has been created](UC_keyset_create.md)
+- **Main success scenario:**
+  - Caller requests to insert a new `Key` with non-cascading name into `KeySet`
+  - If necessary Core resizes `KeySet`
```

Comment:
This says it would not be okay for an implementation to fail after inserting a fixed number of keys. Probably okay to require this, as nowadays also most embedded systems have proper memory allocation.

Reply:
Not sure what you mean here, a functioning `malloc` is a requirement for all of Elektra.